packages/deployer/src/compiler.ts

```diff
@@ -207,6 +237,36 @@ export class Compiler {
         const sourceMapRuntime = compiled.contracts[contractIdentifier].srcmapRuntime;
         const sources = _.keys(compiled.sources);
         const updated_at = Date.now();
+
+        // There is no function overloading in typescript, so we must change the Typescript ABI interface.
+        // Overloaded function names are incremented as follows: functionName, functionName_2, functioname_3, ...
+        // If functionName_N already exists then compilation will fail.
+        const functionList:FunctionList = {};
+        for(let i = 0; i < abi.length; ++i) {
+            const type:string = abi[i].type;
+            if(type !== "function") continue;
```

Comment:
It's time we move the `AbiType` type currently in `@0xproject/abi-gen` to the `@0xproject/types` package. Use it here instead of using a string literal.